config/300-ibmmqsource.yaml

```diff
@@ -162,9 +162,60 @@ spec:
                     oneOf:
                     - required: [value]
                     - required: [valueFromSecret]
-                required:
-                - username
-                - password
+                  tls:
+                    description: TLS configuration for IBM MQ client. For more information, please refer to the IBM MQ
+                      documentation at
+                      https://www.ibm.com/docs/en/ibm-mq/9.2?topic=mechanisms-tls-security-protocols-in-mq.
+                    type: object
+                    properties:
+                      cipher:
+                        description: Message encryption and authentication algorithm.
+                          For more information, please refer to the IBM MQ documentation at
+                          https://www.ibm.com/docs/en/ibm-mq/9.2?topic=tls-cipherspecs-ciphersuites.
+                        type: string
+                      clientAuthRequired:
+                        description: Parameter to specify whether TLS client authentication is required. For more
+                          information, please refer to the IBM MQ documentation at
+                          https://www.ibm.com/docs/en/ibm-mq/9.2?topic=fields-sslclientauth-mqlong.
+                        type: boolean
+                      keyRepository:
+                        description: Key repository for (m)TLS connection. For more information, please refer to the
+                          IBM MQ documentation at https://www.ibm.com/docs/en/ibm-mq/9.2?topic=mq-ssltls-key-repository.
+                        type: object
+                        properties:
+                          kdbSecret:
```

Comment:
opinion, dismiss if you think otherwise:

I am more used to see the `secretKeyRef` name when referencing secrets. Something like

```yaml
keyDatabase:
  secretKeyRef:
     ...
```

Reply:
Decided to go with the `valueFromSecret` in 86af0b1 to have all secret references in the same style in this CRD.